src/promptflow-tools/promptflow/tools/template_rendering.py

```diff
@@ -1,9 +1,31 @@
 # Avoid circular dependencies: Use import 'from promptflow._internal' instead of 'from promptflow'
 # since the code here is in promptflow namespace as well
 from promptflow._internal import tool
-from promptflow.tools.common import render_jinja_template
+from promptflow.tools.common import (
+    render_jinja_template,
+    PromptResult,
+    Escaper,
+    INPUTS_TO_ESCAPE_PARAM_KEY
+)
 
 
 @tool
-def render_template_jinja2(template: str, **kwargs) -> str:
-    return render_jinja_template(template, trim_blocks=True, keep_trailing_newline=True, **kwargs)
+def render_template_jinja2(template: str, **kwargs) -> PromptResult:
+    rendered_template = render_jinja_template(template, trim_blocks=True, keep_trailing_newline=True, **kwargs)
```

Comment:
add comment to explain what this rendering is used for.
something like this, 
```
# step 1: set original string of prompt result.
```

Reply:
added